src/cache/axios.ts

```diff
@@ -30,7 +30,7 @@ export interface CacheAxiosResponse<R = any, D = any> extends AxiosResponse<R, D
    *
    * @see https://axios-cache-interceptor.js.org/config/response-object#id
    */
-  id: string;
+  id?: string;
```

Comment:
same here : that is exactly what breaks the compatibility with axios 1.7.8

Reply:
By the way, I modified the test to assess this `id` is present.

Reply:
but where does it break the compatibility? that's where you need to fix it, an interface cannot block its extension to add new fields.

Reply:
You are right, the problem appears when *using* this interface, after removing the optional flag, here is the transpiler message :
```text
The types of 'interceptors.response.use' are incompatible between these types.
    Types of parameters 'onFulfilled' and 'onFulfilled' are incompatible.
      Type 'AxiosResponse<any, any> | Promise<AxiosResponse<any, any>>' is not assignable to type 'CacheAxiosResponse<any, any> | Promise<CacheAxiosResponse<any, any>>'.
        Type 'AxiosResponse<any, any>' is not assignable to type 'CacheAxiosResponse<any, any> | Promise<CacheAxiosResponse<any, any>>'.
          Type 'AxiosResponse<any, any>' is missing the following properties from type 'CacheAxiosResponse<any, any>': id, cached
```

because of the following :
```TypeScript
export interface AxiosCacheInstance extends CacheInstance, AxiosInstance {
  /* ... */
  interceptors: {
    request: AxiosInterceptorManager<InternalCacheRequestConfig>;
    response: AxiosInterceptorManager<CacheAxiosResponse>;
  };
```

I noticed that in the `InternalCacheRequestConfig`, `id` and `cache` were optional, I assumed it was to solve the same issue.

But I get your point : if anybody relies on the `CacheAxiosResponse` interface, this represents a breaking change. However, since it is not explicitly documented, I assumed it was OK.

Reply:
Let me review my implementation and see if I can create an `InternalCacheAxiosResponse`

Reply:
Actually, reviewing the implementation, theCacheRequestConfig uses optional on `id` and `cache` as documented here : https://axios-cache-interceptor.js.org/config/request-specifics

Reply:
I pushed a new version that keeps backward compatibility *but* it requires many changes across the code. The most important is to ensure that, in the end, the `onFulfilled` receives the `CacheAxiosResponse` response... which is the case.

I am not able to evaluate if this change can break anybody implementation (do people rely on the `AxiosCacheInstance` interface ?), but it works on my use.